src/disk.hpp

```diff
@@ -193,8 +193,9 @@ struct FileDisk {
 #endif
         // Seek and write from memcache
         uint64_t amtwritten;
+        bool errorOnWrite;
```

Comment:
instead of introducing a new variable, you could also set `bReading = true` on line 219 (222 in your patch).

I think regardless this warrants a comment to remind readers that the write position is unspecified when an error occurs

Reply:
I think errorOnWrite needs to be initialized here, since it is checked in the conditional below